Automattic/wp-calypso
# Jetpack Manage: Pricing page - creating plan and product cards

Related to https://github.com/Automattic/jetpack-manage/issues/247

## Proposed Changes

* Currently this PR creates plan cards for the new `manage/pricing` page, as well as product cards, Woo cards and Backup add-on cards.
* These are not yet connected to a license / bundle selector, so a figure of 1 is hard-coded in while developing.
* The savings line in each card currently says 'per bundle per month' for bundles, instead of per license (as the design specs suggest). This can be changed but I've left it as is for now.
* The Get button should directly open the license lightbox on the issue licenses page for the relevant product / including the relevant bundle size. The functionality is only for logged in users at the moment, and not for bundles larger than 1. 
* See the 'for later PRs' below for functionality that doesn't yet work as expected.



### For later PRs:

* The Get button functionality for non-logged in users needs to be added: https://github.com/Automattic/jetpack-manage/issues/259
* The Get button needs to work for bundles larger than 1: https://github.com/Automattic/jetpack-manage/issues/264
* Get buttons for multi select options lead to an error on first load only: https://github.com/Automattic/jetpack-manage/issues/267
* The 'More about' link (lightbox component) is still to be added: https://github.com/Automattic/jetpack-manage/issues/258
* Console errors exist when changing the plan type using the multi-select options (the Security plan card, VaultPress Backup product card): https://github.com/Automattic/jetpack-manage/issues/265
* Styling tweaks are needed to match design specs: https://github.com/Automattic/jetpack-manage/issues/266

## Testing Instructions

* To test this out, apply this PR and visit `http://jetpack.cloud.localhost:3000/manage/pricing`. You should see a Security 10GB and Complete card
* Try toggling between the 10GB and 1TB option in the Security plans card (and VaultPres Backup products card) - the content should change accordingly.
* You can change the bundleSize in the `LicenseProductsList` component (in `jetpack-cloud/sections/manage/pricing/primary/index.tsx`) to 5, and the product name in the Security card should change, and the prices in both cards. These will all match what you see on the `partner-portal/issue-licenses` page.
  * You may need to modify the bundleSize in `most-popular-plans.tsx` and `all-license-items.tsx` if the bundleSize isn't being picked up initially.
  * To test the 'Get' button, hover over any of the Get buttons and as a logged in partner you should see a URL similar to `https://cloud.jetpack.com/partner-portal/issue-license?product_slug=jetpack-anti-spam&bundle_size=5&source=manage-pricing-page`. As a logged in non partner the same button should now direct you to `https://cloud.jetpack.com/manage/signup?issue-license=yes&product_slug=jetpack-anti-spam&bundle_size=5&source=manage-pricing-page`.
    * The Get button link should also directly open up the 'review license' lightbox when directly clicking on it.
    * Note that this only works when the quantity is 1 (not yet setup for larger quantities - see follow up PRs) - nor on first load for variants (eg backup and security 10GB and 1TB).

~Edit to add - to be able to see plans and products, you'll need to make sure you have a partner account and also that a key is selected - you may need to visit and revisit the partner portal dashboard to make sure a key is selected and the sites dashboard loads as usual - then the pricing page will work. This is only while using the auth endpoint.~ This isn't needed anymore as we're using the public endpoint now.

Screenshot showing all product and plan card component additions from this PR:

<img width="452" alt="Screenshot 2024-01-31 at 12 24 10" src="https://github.com/Automattic/wp-calypso/assets/16754605/d047d350-6f08-407a-a62b-9b2d2bab762f">


## Pre-merge Checklist
